test/vvenc_unit_test/vvenc_unit_test.cpp

```diff
@@ -113,6 +113,30 @@ static inline bool compare_values_2d( const std::string& context, const T* ref,
   return true;
 }
 
+template<typename T>
+class FixedInputGenerator
+{
+public:
+  explicit FixedInputGenerator( T val ) : m_val( val )
```

Comment:
I dont like the idea of a fixed value generator. If you really want to test overflows, I'd say doing patterns of min/max values would be more suitable (as the search often does operation on differences of values). Maybe a min-max generator would be more interesting, that randomly selects either the min or max values.